src/components/renderer.cpp

```diff
@@ -767,9 +755,14 @@ bool renderer::on(const signals::parser::reverse_colors&) {
   return true;
 }
 
-bool renderer::on(const signals::parser::offset_pixel& evt) {
-  m_log.trace_x("renderer: offset_pixel(%f)", evt.cast());
-  m_blocks[m_align].x += evt.cast();
+bool renderer::on(const signals::parser::offset& evt) {
+  m_log.trace_x("renderer: offset(%f)", evt.cast());
+
+  auto& x_offset = m_blocks[m_align].x;
+  auto offset_width = unit_utils::geometry_to_pixel(unit_utils::geometry_from_string(evt.cast()), m_bar.dpi_x);
+
+  draw_offset(x_offset, offset_width);
+  x_offset += offset_width;
```

Comment:
Using references like this can be a bit confusing. I think it is clearer if we access and modify `m_blocks[m_align].x` directly.

Reply:
I agree I don't know why I did that. This is ugly :scream:

Reply:
I agree I don't know why I did that. This is ugly :scream:

Reply:
Fixed by https://github.com/jaagr/polybar/pull/1671/commits/18dfa8ddaf802b1c62646d405cadccf59d6da13d